src/libtiled/tilelayer.cpp

```diff
@@ -170,8 +170,15 @@ void Tiled::TileLayer::setCell(int x, int y, const Cell &cell)
 {
     Q_ASSERT(contains(x, y));
 
-    if (cell.isEmpty() && !findChunk(x, y))
-        return;
+    if (!findChunk(x, y)) {
+        if (cell.isEmpty())
+            return;
+        else
+            mBounds = mBounds.united(QRect(x - (x & CHUNK_MASK),
+                                           y - (y & CHUNK_MASK),
+                                           CHUNK_SIZE,
+                                           CHUNK_SIZE));
```

Comment:
if the line in the else is multiple lines long please use `{}` around it. And since the `else` then has it, the `if`-part also should have it